src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -2655,25 +2655,24 @@ def update_artifact(
                 session=session,
             )
 
-            # Handle tag updates.
-            if artifact_update.add_tags:
-                self._attach_tags_to_resource(
-                    tag_names=artifact_update.add_tags,
-                    resource=existing_artifact,
-                    session=session,
-                )
-            if artifact_update.remove_tags:
-                self._detach_tags_from_resource(
-                    tag_names=artifact_update.remove_tags,
-                    resource=existing_artifact,
-                    session=session,
-                )
-
             # Update the schema itself.
             existing_artifact.update(artifact_update=artifact_update)
             session.add(existing_artifact)
             session.commit()
             session.refresh(existing_artifact)
+
+            # Handle tag updates.
+            self._attach_tags_to_resources(
```

Comment:
Now that this was moved after the `session.refresh(existing_artifact)`, does the `update_artifact` method actually return the artifact with the updated tags?

Reply:
Good catch, I have added a `session.refresh(...)` each time we attach or detach tags to or from a schema.